File: platform/sdk-examples/typescript/logs-operation-example.ts
#!/usr/bin/env node
/**
 * Simple Logs Operation Example
 *
 * This example shows basic log operations:
 * - List available logs
 * - Simple log API usage
 *
 * Usage:
 *     npm run build && node dist/examples/logs-operation-example.js
 *     or
 *     npx tsx examples/logs-operation-example.ts
 *
 * Environment variables required:
 * - RESPAN_API_KEY
 * - RESPAN_BASE_URL (optional)
 */

import { config } from 'dotenv';
import { LogAPI } from '../src/index.js';

// Load environment variables with override
config({ override: true });

async function main() {
  const apiKey = process.env.RESPAN_API_KEY;
  const baseUrl = process.env.RESPAN_BASE_URL;

  if (!apiKey) {
    console.log('❌ RESPAN_API_KEY not found in environment');
    return;
  }

  const logApiClient = new LogAPI({ apiKey, baseUrl });

  try {
    const logList = await logApiClient.list();
    console.log(logList);
  } catch (error) {
    console.error('❌ Error:', error);
  }
}

if (import.meta.url === `file://${process.argv[1]}`) {
  main().catch(console.error);
}

File: platform/sdk-examples/python/logs_operation_example.py
from dotenv import load_dotenv

load_dotenv(override=True)
from respan.logs.api import LogAPI
import os


def main():
    api_key = os.getenv("RESPAN_API_KEY")
    base_url = os.getenv("RESPAN_BASE_URL")

    log_api_client = LogAPI(api_key=api_key, base_url=base_url)

    log_list = log_api_client.list()
    print(log_list)


if __name__ == "__main__":
    main()
